### PR TITLE
RDMA soft_RoCE implement

### PR DESCRIPTION
**Pull request by FlashBarryAllen**

### CODE REVIEW

File: Packet++/src/InfiniBandLayer.cpp [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        
+	}
+
+	std::string InfiniBandLayer::toString() const
+	{
+        std::ostringstream opCodeStream;
+        opCodeStream << getOpcode();
+
+        return "InfiniBand Layer, Opcode: " + opCodeStream.str();
+	}
```
Comment (Dimi1010) [inline Packet++/src/InfiniBandLayer.cpp:58]: Using `std::to_string(x)` when converting single values is more efficient then using a `std::stringstream`.

File: Tests/Packet++Test/TestDefinition.h [hunk 2]
```diff
@@ -262,3 +262,6 @@ PTF_TEST_CASE(Asn1EncodingTest);
 // Implemented in LdapTests.cpp
 PTF_TEST_CASE(LdapParsingTest);
 PTF_TEST_CASE(LdapCreationTest);
+
+// Implemented in InfiniBandTests.cpp
+PTF_TEST_CASE(InfiniBandPacketParsing);
```
Comment (Dimi1010) [inline Tests/Packet++Test/TestDefinition.h:267]: Missing new line at the end of file.
```suggestion
PTF_TEST_CASE(InfiniBandPacketParsing);

```

File: Packet++/header/InfiniBandLayer.h [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		    : Layer(data, dataLen, prevLayer, packet, IB)
+		{}
+
+        InfiniBandLayer( uint8_t opcode, int se,
+                    int mig, int pad, uint16_t pkey, uint32_t qpn, int ack_req,
+                    uint32_t psn);
+
+        rxe_bth* getBthHeader() const
+        {
+            return (rxe_bth*)m_Data;
```
Comment (Dimi1010) [inline Packet++/header/InfiniBandLayer.h:75]: Cpp style casts are preferred, as they are more explicit.
```suggestion
            return reinterpret_cast<rxe_bth*>(m_Data);
```

File: Packet++/src/InfiniBandLayer.cpp [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+    InfiniBandLayer::InfiniBandLayer( uint8_t opcode, int se,
+                    int mig, int pad, uint16_t pkey, uint32_t qpn, int ack_req,
+                    uint32_t psn)
+    {
+        const size_t headerLen = sizeof(rxe_bth);
+		m_DataLen = headerLen;
+		m_Data = new uint8_t[headerLen];
+		memset(m_Data, 0, headerLen);
+		rxe_bth* bthHdr = (rxe_bth*)m_Data;
```
Comment (Dimi1010) [inline Packet++/src/InfiniBandLayer.cpp:19]: ditto, C-cast -> Cpp casts

**Review by Dimi1010 — CHANGES_REQUESTED** [review]
Hello, thanks for your work for this PR. Unfortunately there are a couple of issues with it that need to be addressed.

As per [CONTRIBUTING.md](https://github.com/seladb/PcapPlusPlus/blob/master/CONTRIBUTING.md) all PRs must be merged into `dev` and not `master`. The CI is also configured to run tests on PR's targeting the `dev` branch. I have switched the base of the PR to be `dev` instead, but please pull the changes from the newest `dev` head into your feature branch. It would also allow the CI to update and run the automatic tests against the changes.

I have also included a couple of suggestions to the code, that I noticed while looking through it.

**Comment by tigercosmos** [comment]
@FlashBarryAllen  Could you update the PR description? also, please run `pre-commit` first. You can find more information in [CONTRIBUTING.md](https://github.com/seladb/PcapPlusPlus/blob/master/CONTRIBUTING.md) as @Dimi1010 has mentioned.

**Comment by seladb** [comment]
@FlashBarryAllen do you have time to continue working on this PR?

**Comment by FlashBarryAllen** [comment]
@seladb the next 2 days I have time to implement the left code.

**Pull request closed without merging**